File: java/complete/myretro/src/test/java/com/apress/myretro/MyretroUserJsonTests.java
package com.apress.myretro;

import com.apress.myretro.user.User;
import com.apress.myretro.user.UserRole;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.json.JsonTest;
import org.springframework.boot.test.json.JacksonTester;
import org.springframework.boot.test.json.JsonContent;

import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;

@JsonTest
public class MyretroUserJsonTests {

    @Autowired
    private JacksonTester<User> userJson;

    @Test
    void serialize() throws Exception {
        User user = new User("Dummy","dummy@email.com","","dummyP1ssWd", Arrays.asList(UserRole.USER),true);
        JsonContent<User> result =  this.userJson.write(user);
        assertThat(result).hasJsonPathStringValue("$.name");
        assertThat(result).extractingJsonPathStringValue("$.name").isEqualTo("Dummy");
        assertThat(result).extractingJsonPathStringValue("$.gravatarUrl").isEmpty();
    }

    @Test
    void deserialize() throws Exception {
        String content = "{\"name\":\"Dummy\",\"email\":\"dummy@email.com\",\"gravatarUrl\":\"\",\"password\":\"dummyP1ssWd\",\"userRole\":[\"USER\"],\"active\":true}";
        assertThat(this.userJson.parse(content)).isEqualTo(new User("Dummy","dummy@email.com","","dummyP1ssWd", Arrays.asList(UserRole.USER),true));
        assertThat(this.userJson.parseObject(content).getEmail()).isEqualTo("dummy@email.com");
    }
}

File: kotlin/complete/myretro/src/test/kotlin/com/apress/myretro/MyretroUserJsonTests.kt
package com.apress.myretro

import com.apress.myretro.user.User
import com.apress.myretro.user.UserRole
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.json.JsonTest
import org.springframework.boot.test.json.JacksonTester
import java.util.*

@JsonTest
class MyretroUserJsonTests {

    @Autowired
    private val userJson: JacksonTester<User>? = null

    @Test
    @Throws(Exception::class)
    fun serialize() {
        val user = User("Dummy", "dummy@email.com", "", "dummyP1ssWd", listOf(UserRole.USER), true)
        val result = userJson!!.write(user)
        Assertions.assertThat(result).hasJsonPathStringValue("$.name")
        Assertions.assertThat(result).extractingJsonPathStringValue("$.name").isEqualTo("Dummy")
        Assertions.assertThat(result).extractingJsonPathStringValue("$.gravatarUrl").isEmpty()
    }

    @Test
    @Throws(Exception::class)
    fun deserialize() {
        val content =
            "{\"name\":\"Dummy\",\"email\":\"dummy@email.com\",\"gravatarUrl\":\"\",\"password\":\"dummyP1ssWd\",\"userRole\":[\"USER\"],\"active\":true}"
        Assertions.assertThat(userJson!!.parse(content))
            .isEqualTo(User("Dummy", "dummy@email.com", "", "dummyP1ssWd", listOf(UserRole.USER), true))
        Assertions.assertThat(userJson.parseObject(content).email).isEqualTo("dummy@email.com")
    }
}